### PR TITLE
Include release publish workflow and improve daily builds

### PR DESCRIPTION
**Pull request by keizer619**
## Purpose

- Archive daily builds
- Include release publish workflow

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+name: Publish release
+
+on:
+  workflow_dispatch:
+
+jobs:
+  publish-release:
+    name: Publish Release
+    runs-on: ubuntu-latest
```
Comment (pubudu91) [inline .github/workflows/publish-release.yml:9]: Shall we stick to a specific version? Recently we ran in to a weird build failure which turned out to be due to `ubuntu-latest` pulling in 20.04 (it used pull 18).

Comment (keizer619) [reply .github/workflows/publish-release.yml:9]: In which repo it happened? is it during release? 

Apparently most of repos are using `ubuntu-latest` including [ballerina-lang](https://github.com/ballerina-platform/ballerina-lang/blob/master/.github/workflows/publish_release.yml#L8)

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set version env variable
+        id: version-set
+        run: |
+          SHORT_VERSION=$((grep -w 'version' | cut -d= -f2) < gradle.properties
+          RELEASE_VERSION=$((grep -w 'shortVersion' | cut -d= -f2) < gradle.properties | rev | cut --complement -d- -f1 | rev)
```
Comment (pubudu91) [inline .github/workflows/publish-release.yml:16]: Shouldn't these two be swapped? i.e., `RELEASE_VERSION <= 'version'` and `SHORT_VERSION <= 'shortVersion'`?

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (Kishanthan) [reply .github/workflows/publish-release.yml:9]: +1

Comment (Kishanthan) [reply .github/workflows/publish-release.yml:9]: Happened for website repo, with some stuffs related to python and aws cli tools

Comment (keizer619) [reply .github/workflows/publish-release.yml:9]: In that case shall we leave `ubuntu-latest` for these and keep exact version which ever failing ones.
Usually python versions in Ubuntu 18 and 20 are different and our python script may not support the latest. Ideally we should update that script to support latest

**Review by pubudu91 — APPROVED** [review]
Looks ok to me.

**Pull request merged**